### PR TITLE
Added pyswmm simulation class; restructured and added descriptors

### PR DESCRIPTION
**Pull request by bemcdonnell**
@goanpeca, I made some pretty significant changes here.  Please review the new pyswmm.Simulation class and Links.py link classes. Descriptors have been incorporated to *set* different parameters. 

I updated all of the testing files as well so I'm sorry for all the noise

### CODE REVIEW

File: test/TestModel10_simulationItermethods.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+'''
```
Comment (bemcdonnell) [inline test/TestModel10_simulationItermethods.py]: @michaeltryby,  The following illustrates three approaches to run a simulation with pyswmm.

File: pyswmm/pyswmm.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
+"""Base class for a SWMM Simulation.
```
Comment (goanpeca) [inline pyswmm/pyswmm.py]: """Base class for a SWMM Simulation."""

File: pyswmm/pyswmm.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+"""
+
+from swmm5 import PYSWMM
+
+
+__author__ = 'Bryant E. McDonnell (EmNet LLC) - bemcdonnell@gmail.com'
+__copyright__ = 'Copyright (c) 2016 Bryant E. McDonnell'
+__licence__ = 'BSD2'
+__version__ = '0.2.1'
```
Comment (goanpeca) [inline pyswmm/pyswmm.py]: All these variables should be moved to the `pyswmm/__init__.py`

File: pyswmm/pyswmm.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        if time <= 0.0:
+            self._model.swmm_end()
+            raise StopIteration
+        return self._model
+
+    def __exit__(self, *a):
+        self._model.swmm_end()
+        self._model.swmm_close()
+
+    def step_advance(self,advance_seconds):
```
Comment (goanpeca) [inline pyswmm/pyswmm.py]: `self, advance_seconds`  space

File: pyswmm/pyswmm.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     for ind, step in enumerate(sim):
+        ...         pass
+        ...     sim.report()        
+        self._model.swmm_report()
+        """
+
+    def close(self):
+        self.__exit__()
```
Comment (goanpeca) [inline pyswmm/pyswmm.py]: no docstring?

File: test/TestModel1_CurrentTime.py [hunk 6]
```diff
@@ -13,10 +13,10 @@
 #point to location of the pyswmm file
 sys.path.append(os.getcwd()+'\\..\\pyswmm\\')
 
-from swmm5 import pyswmm
+from swmm5 import PYSWMM
```
Comment (goanpeca) [inline test/TestModel1_CurrentTime.py]: The object should probably be called

`PySWMM`

File: pyswmm/links.py [hunk 7]
```diff
@@ -0,0 +1,693 @@
+from toolkitapi import *
+from swmm5 import SWMMException, PYSWMMException
+
+__author__ = 'Bryant E. McDonnell (EmNet LLC) - bemcdonnell@gmail.com'
+__copyright__ = 'Copyright (c) 2016 Bryant E. McDonnell'
+__licence__ = 'BSD2'
+__version__ = '0.2.1'
+
```
Comment (goanpeca) [inline pyswmm/links.py]: All this should go into the `pyswmm/__init__.py`

File: pyswmm/links.py [hunk 8]
```diff
@@ -0,0 +1,693 @@
...
+        return self
+    
+    def next(self):
+        if self._cuindex < self._nLinks:
+            linkobject = Link(self._model, self.linkid)
+            self._cuindex+=1 #Next Iteration
+            return linkobject
+        else:
+            raise StopIteration()        
+    @property
```
Comment (goanpeca) [inline pyswmm/links.py]: space between lines

File: pyswmm/links.py [hunk 9]
```diff
@@ -0,0 +1,693 @@
...
+    def linkid(self):
+        "Link ID"
```
Comment (goanpeca) [inline pyswmm/links.py]: """Link ID."""

File: pyswmm/links.py [hunk 10]
```diff
@@ -0,0 +1,693 @@
...
+        return self._model.getObjectId(ObjectType.LINK,self._cuindex)
```
Comment (goanpeca) [inline pyswmm/links.py]: `return self._model.getObjectId(ObjectType.LINK, self._cuindex)`

File: pyswmm/links.py [hunk 11]
```diff
@@ -0,0 +1,693 @@
...
+        ...     print link.q0
+        ...     link.offset1 = 0.2
+        ...     print link.offset1
+        >>> 0.1
+        >>> 0.2
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.q0, param)
+    
+    @property
+    def qlimit(self):
```
Comment (goanpeca) [inline pyswmm/links.py]: these names are weird :-|

limit.. q0 ... offset1 etc....

I think we can use much better names,

File: pyswmm/links.py [hunk 12]
```diff
@@ -0,0 +1,693 @@
...
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.clossave
+        >>> 0
+        """
+        return self._model.getLinkParam(self._linkid,LinkParams.cLossAvg)
+
+    @clossave.setter
+    def clossave(self, param):
```
Comment (goanpeca) [inline pyswmm/links.py]: these names are hard to understand...

`average_head_loss` is already more clear than `clossave`

File: pyswmm/links.py [hunk 13]
```diff
@@ -0,0 +1,693 @@
...
+        ...     print link.seepagerate
+        ...     link.seepagerate = 0.2
+        ...     print link.seepagerate
+        >>> 0
+        >>> 0.2  
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.seepRate, param)
+        
+    @property
+    def flow(self):
```
Comment (goanpeca) [inline pyswmm/links.py]: For instance, flow :-), nice clear name!

File: pyswmm/links.py [hunk 14]
```diff
@@ -0,0 +1,693 @@
...
+        >>> 0
+        >>> 1.2
+        >>> 1.5
+        >>> 1.9
+        >>> 1.2
+        """  
+        return self._model.getLinkResult(self._linkid,LinkResults.newFlow)
+    
+    @property
+    def depth(self):
```
Comment (goanpeca) [inline pyswmm/links.py]: awesome name!

File: pyswmm/links.py [hunk 15]
```diff
@@ -0,0 +1,693 @@
...
+        >>> 0
+        >>> 1.2
+        >>> 1.5
+        >>> 1.9
+        >>> 1.2
+        """  
+        return self._model.getLinkResult(self._linkid,LinkResults.newVolume)
+    
+    @property
+    def usxsectionarea(self):
```
Comment (goanpeca) [inline pyswmm/links.py]: `ups_xsection_area` ? we can go over what names and conventiones should we use to make them short enough but understandable enough

File: pyswmm/links.py [hunk 16]
```diff
@@ -0,0 +1,693 @@
...
+        >>> 0
+        >>> 1.2
+        >>> 1.5
+        >>> 1.9
+        >>> 1.2
+        """  
+        return self._model.getLinkResult(self._linkid,LinkResults.surfArea2)
+    
+    @property
+    def currentsetting(self):
```
Comment (goanpeca) [inline pyswmm/links.py]: current_setting

File: pyswmm/links.py [hunk 17]
```diff
@@ -0,0 +1,693 @@
...
+        >>> 0
+        >>> 1
+        >>> 0
+        >>> 0.5
+        >>> 1
+        """  
+        return self._model.getLinkResult(self._linkid,LinkResults.setting)
+    
+    @property
+    def targetsetting(self):
```
Comment (goanpeca) [inline pyswmm/links.py]: target_setting

**Comment by bemcdonnell** [comment]
@goanpeca I addressed all the comments.  For sake of moving this project forward, I am going to go ahead and merge these changes over and i'll cascade them back up to the master branch.

**Pull request merged**